### ISSUE TITLE
Why my code get better f1_score on older versions

### ISSUE THREAD

**Issue by DSOTM-pf** [keras-team/tf-keras#374]
**System information**.
- Have I written custom code (as opposed to using a stock example script provided in Keras): No
- OS Platform and Distribution:  Linux Ubuntu 16.04
- TensorFlow installed from binary
- TensorFlow version (use command below): 2.10.0 , 2.7.0,
- Python version: 3.7.10
- GPU model and memory: TITAN V

**Describe the problem**.
Hi, I adopt different Tensorflow and Keras versions with my code. But it returns a better f_1 Score on older versions. 

To avoid the effect of random seeds, I try to set the random seed using this approach.
https://github.com/keras-team/keras/issues/2743#issuecomment-904169662
```python
import random
random.seed(2022) 
np.random.seed(2022)  
tf.random.set_seed(2022) 
```
I also added ```shuffle = False in model.fit()```

**Describe the current behavior**.
The reproduction results are as follows:
|Tensorflow|Keras|F1 Score|
|--|--|--|
|2.10.0|2.10.0|0.5369807497467073|
|2.7.0|2.7.0|0.29846938775510207|
|2.4.0|2.4.0|0.7376928728875827|

**Describe the expected behavior**.
Why the new version has worse results? The latest version of the score should be better.

**Standalone code to reproduce the issue**.
I post the code on colad.
[click me to reproduce](https://colab.research.google.com/drive/1BWZMI7uPmYfOgobBHRH0H2q3WXOSmdDn?usp=sharing)

[click me to download dataset](https://github.com/tensorflow/tensorflow/files/9929997/dataset.zip)
